uiowa/uiowa
# Clean up old media revisions

Resolves #4811 

Previous work in #4641 prevents new media revisions from being created when media entities are edited and saved, including when new files are attached. Media entities made prior to that PR's deployment may have multiple revisions, including revisions using different files as in the case of file overrides.

This PR cleans up any old revisions if they exist for all media entities. Part of this process should also mark any files which were previously only associated with old revisions as no longer in use (tracked in the `file_usage` table). These files should be marked status = 0 as part of this update, which would allow cron processes to clean them up at a later time as no longer used by the site.





# How to test



- Sync a site, check that the update hook runs without issue
- Check that the media entities count does not change (number of rows in the `media` table) following the update hook
- Check that the number of rows in the `media_revision` table equals the number of rows in the `media` table (it may have been larger prior to the update hook)
- Check if the number of rows in the `file_usage` table has changed
  - If it has changed, check that the difference is equal to the number of files in `file_managed` which have a status of '0'
  - Of the files in `file_managed` which have a status of 0, check that they are associated to media entities which have new files associated with their most current revisions
